indigo/indigo/src/main/scala/indigo/shared/input/Mouse.scala

```diff
@@ -47,6 +48,20 @@ final class Mouse(
       case _                                             => false
     }
 
+  private def wasScrolled: Boolean = mouseEvents.exists {
+    case _: MouseEvent.Wheel => true
+    case _                   => false
+  }
+
+  lazy val scrolled: Option[MouseWheelDir] =
+    if wasScrolled then
+      val amount = mouseEvents.foldLeft(0d) {
+        case (acc, MouseEvent.Wheel(_, deltaX)) => acc + deltaX
+        case (acc, _)                           => acc
+      }
+      if amount < 0 then Some(MouseWheelDir.Up) else Some(MouseWheelDir.Down)
+    else Option.empty[MouseWheelDir]
```

Comment:
Probably fine because it's probably only a handful of events and maybe 2 wheel events per frame, so this is just for future reference: A general performance point in the engine is to avoid `foldLeft` on `List` as I've found it to be very slow.

Reply:
I think the first check is unnecessary, and possibly amounts to iterating over the list anyway?

```scala
  lazy val scrolled: Option[MouseWheelDir] =
    val amount = mouseEvents.foldLeft(0d) {
      case (acc, MouseEvent.Wheel(_, deltaX)) => acc + deltaX
      case (acc, _)                           => acc
    }
    if amount == 0 then None
    else if amount < 0 then Some(MouseWheelDir.Up)
    else Some(MouseWheelDir.Down)
```

I don't feel strongly about it, just trying it out to see how it feels.